### ISSUE TITLE
Request textDocument/hover failed. On VSCode

### ISSUE THREAD

**Issue by cl0v** [dart-lang/sdk#47736]
This error is happaning to me on VsCode:

Notification:
A request has failed. See the output for more information.

Output console:
[Error - 11:18:06 PM] Request textDocument/hover failed.
Message: Invalid file path
Code: -32003
.... [any .dart file path that im trying to work here]

Dart version : v3.19.2

**Comment by Elietshi1004** [comment]
what's your flutter version?

Le ven. 19 nov. 2021 à 00:32, Marcelo Viana ***@***.***> a
écrit :

**Comment by tomasbaran** [comment]
Please reopen who has the power. This is a HUGE bug. I wasted 2.5 days on this issue which actually didn't even let me 'flutter build ipa' even though the app worked in simulator. I'm not sure whether it's only this issue or there are others what I can say though that only downgrading from the stable 2.5.3 to 2.2.1 helped. :(

**Comment by GabrielUlisses** [comment]
On vs code, press F1 and type: dart: restart analysis server
It's work to me

**Comment by SakshamKarnawat** [comment]
Dart: Restart Analysis Server isn't working for me. Still getting this extremely annoying output. Stuck on a project since 3 days because of this.

**Comment by casvanluijtelaar** [comment]
> On vs code, press F1 and type: dart: restart analysis server It's work to me

> Hmm yeah okay that might work for you but just like the original answer "just restart dart analysis server" that is clearly not a fix for this issue / bug

**Comment by GabrielUlisses** [comment]
I had this problem again even after restart dart analysis server, I copied the contents of the file with error to another document and the problem persisted, later I rewrote all the content manually in a new document and everything was resolved. I had the same problem in another project but I only needed to erase and rewrite a line of code, I believe it might be an error when interpreting a specific character.

I hope this review is helpful.

**Comment by srawlins** [comment]
Thanks for all of the reports, everyone. In order to debug this issue, please include a minimal reproduction of code which causes textDocument/hover to fail.

If anyone has a crash which is unrelated, you are also free to file a new issue with the bug report. We don't really triage ongoing discussion on closed issues.

**Comment by SakshamKarnawat** [comment]
> Dart: Restart Analysis Server isn't working for me. Still getting this extremely annoying output. Stuck on a project since 3 days because of this. 
> Please reopen this issue. It isn't solved by any means.

Update:

After three days of trying everything, reinstalling all extensions, reinstalling VSCode, redownloading Flutter SDK, I finally tried removing specific functions, widgets, strings, etc. one by one until I found the bit of code that was producing the error output. In my case, it was a string with multiple special characters. Finally solved the issue.

Hope this helps someone.

**Comment by srawlins** [comment]
That is a great clue. We might need to know what types of special characters in order to make progress...

**Comment by SakshamKarnawat** [comment]
I was using a RegExp to validate email address so it had many special characters. There were no errors shown in the file so I'm not sure which characters caused it but I believe it is probably one of these:    #    $   %   &

**Comment by UlasKasim** [comment]
I have the same error message on vs code
`[Error - 22:24:34] Request textDocument/codeAction failed.
  Message: Invalid file path
  Code: -32003 `

When I look into 'Help->ToogleDeveloperTools' I see this:
`Invalid file path: Error: Invalid file path
	at handleResponse (c:\Users\*\.vscode\extensions\dart-code.dart-code-3.29.0\out\dist\extension.js:27900:48)
	at processMessageQueue (c:\Users\*\.vscode\extensions\dart-code.dart-code-3.29.0\out\dist\extension.js:27715:17)
	at Immediate.<anonymous> (c:\Users\*\.vscode\extensions\dart-code.dart-code-3.29.0\out\dist\extension.js:27699:13)
	at processImmediate (internal/timers.js:461:21)`

So, I checked the location and there is no 'dist' folder. I don't know the solution but maybe someone with knowledge can help

**Comment by mustafatatarhan** [comment]
Try to delete the codes in the .dart file that has the error. write it line by line from the beginning, and when you will face this problem go to it's main file of the variable/class/method who causing the problem  and check import lines. You will see at least one file path that does not appear to be incorrect, but is incorrect. Correct it. And its over.  I followed the same steps and fixed the import error and found the solution. I hope I explained my problem and solution well and it will be useful to someone.

**Comment by mustafatatarhan** [comment]
Everyone has faced this problem in different situations. I faced this error while copying files from my clone application to my main application. I'm sharing my solution in case anyone encounters the same error in the same situation :

Try to delete the codes in the .dart file that has the error. write it line by line from the beginning, and when you will face this problem go to it's main file of the variable/class/method who causing the problem  and check import lines. You will see at least one file path that does not appear to be incorrect, but is incorrect. Correct it. And its over.  I followed the same steps and fixed the import error and found the solution. I hope I explained my problem and solution well and it will be useful to someone.

**Comment by Praful** [comment]
I had this error when I clicked on a stacktrace in a VS Code PowerShell terminal. Doing Ctrl+click in the Terminal opened the file I was editing (and where the error was). However, I now has the same file opened twice each in its own tab. The second version had the full path and highlighting anything caused the "Invalid file path" error. So I closed that newly opened file and went back the other tab. The error disappeared.

In summary close any files that you've opened from a VS Code Terminal and open from the Explorer view instead.

HTH someone.

**Comment by clebemachado** [comment]
Is there any chance this problem is in vsCode? In android studio and intelij the same error did not occur.

**Comment by eseidel** [comment]
OK, open this file: https://github.com/RubberDuckEng/html6/blob/2f1027fc2091311f8ddcef3a24a853209ef238b8/bin/run_html5_lib_tests.dart

I'm on:
Version: 1.65.2 (user setup)
Commit: c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1
Date: 2022-03-10T14:33:55.248Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

amodio.tsl-problem-matcher@0.5.0
auchenberg.vscode-browser-preview@0.7.2
bungcip.better-toml@0.3.2
cschlosser.doxdocgen@1.4.0
Dart-Code.dart-code@3.36.0
Dart-Code.flutter@3.36.0
dbaeumer.vscode-eslint@2.2.2
dtsvet.vscode-wasm@1.3.1
FelixAngelov.bloc@6.5.0
GitHub.vscode-pull-request-github@0.38.1
gurumukhi.selected-lines-count@1.4.0
jeff-hykin.better-cpp-syntax@1.15.13
JuanBlanco.solidity@0.0.137
matklad.rust-analyzer@0.2.964
ms-azuretools.vscode-docker@1.20.0
ms-dotnettools.csharp@1.24.1
ms-python.python@2022.2.1924087327
ms-python.vscode-pylance@2022.3.1
ms-toolsai.jupyter@2022.2.1030672458
ms-toolsai.jupyter-keymap@1.0.0
ms-toolsai.jupyter-renderers@1.0.6
ms-vscode-remote.remote-containers@0.224.2
ms-vscode-remote.remote-ssh@0.76.1
ms-vscode-remote.remote-ssh-edit@0.76.1
ms-vscode-remote.remote-wsl@0.64.2
ms-vscode.cmake-tools@1.9.2
ms-vscode.cpptools@1.8.4
ms-vscode.cpptools-extension-pack@1.1.0
ms-vscode.cpptools-themes@1.0.0
ms-vsliveshare.vsliveshare@1.0.5418
ms-vsliveshare.vsliveshare-audio@0.1.91
nelarius.vscode-wren@0.1.1
rust-lang.rust@0.7.8
stkb.rewrap@1.16.3
techer.open-in-browser@2.0.0
Tobiah.unity-tools@1.2.12
twxs.cmake@0.0.17
uctakeoff.vscode-counter@3.0.3
Unity.unity-debug@3.0.2
vadimcn.vscode-lldb@1.6.10

**Comment by Shahin-RockstarIT** [comment]
`[Error - 5:35:16 PM] Request textDocument/hover failed.
  Message: Invalid file path
  Code: -32003 `

Still not solved???

**Comment by clebemachado** [comment]
For me it was the vscode that was corrupted.

**Comment by Babwenbiber** [comment]
For me it was a problem of dependencies. I had some self implemented packages that I edited and I had to call the build runner and pub get again.

**Comment by Leuow95** [comment]
I was having the same issue. 
With an implementation of localizations, I don't know why, but a specific StringResource was causing the problem.
To solve, I changed the order of the StringResources and rewrote the specific lines.

**Comment by NUACHE** [comment]
> in VSCode just open the command pallet and type the command > dart: restart analysis server

worked for me.

**Comment by klausjakobsen** [comment]
> > in VSCode just open the command pallet and type the command > dart: restart analysis server
> 
> worked for me.

Same here

**Comment by ashraf267** [comment]
I think this error can be caused by different things. For me, I noticed the error immediately I deleted the test folder in my flutter app. So just to be sure, I closed the project and created a new dummy flutter project in vscode and the error didn't appear anymore. So if it happens you deleted your test folder, mistakenly or not, try retrieving it or copying the test folder from another project and put it back or creating an entirely new project.

**Comment by vietstone-ng** [comment]
> `[Error - 5:35:16 PM] Request textDocument/hover failed. Message: Invalid file path Code: -32003 `
> 
> Option 1: in VSCode just open the command pallet and type the **command** `dart: restart analysis server`
> 
> Option 2: `brew reinstall --cask visual-studio-code`

Option 1 doesn't work for me.

Option 2 works!! Thank you

**Comment by adonig** [comment]
I had the same issue with Pyright in VSCodium and I was able to fix it by uninstalling and reinstalling the Pyright extension.

**Comment by DanTup** [comment]
I just stumbled across this issue. Is anyone still seeing this on a recent SDK?

@eseidel 

> Message: URI was not an absolute file path (missing drive letter)
Code: -32003
> 
> This is after having opened the document via a stacktrace seen when running the script from powershell within VSC.

This one was an issue where VS Code could open files without drive letters (a path relative to the drive of the "current" directory for the VS Code process) and then they'd be sent to the LSP server (Which doesn't know which drive letter is the "current drive letter according to VS Code).

That was fixed (https://github.com/Dart-Code/Dart-Code/issues/3922) by having the Dart extension detect the paths itself instead of relying on VS Code, so that should be fixed (if not, please file an issue in Dart-Code).

If anyone anyone is still seeing this, please let me know (and if you can provide any concrete steps to reproduce, that would certainly help!).

**Comment by irajeshh** [comment]
SOLVED by [deleting ~/.dartServer](https://github.com/dart-lang/sdk/issues/51455#issuecomment-1446564223) folder which was located like shown in the image.
![image](https://github.com/dart-lang/sdk/assets/85046336/cbbb3848-e18f-49eb-879e-9eb08178a48e)


I've tried the below:
Upgrading flutter.
Uninstalling flutter, dart plugins.
Uninstalling vscode.
Clean start flutter installation from scratch.

**Comment by SakshamKarnawat** [comment]
Sometimes the quotes (" or ')  cause this issue, when copied from another app like WhatsApp, Slack, etc., since it's not a valid quote character according to VSCode. Removing those quote symbols and typing them again might fix the issue. Hope this helps someone.

**Comment by DanTup** [comment]
@SakshamKarnawat if you have an example that reproduces this, please do file a new issue so it can be fixed. Thanks!

**Comment by kousik-patra** [comment]
I fixed it by just deleting the file that caused this error and creating it afresh.
